### ISSUE TITLE
How can I use "yyyy-MM-dd HH" inside inputDateFormat in XYChartScrollbar ?

### ISSUE THREAD

**Issue by siruku6** [amcharts/amcharts4#3068]
**Question**

I used XYChartScrollbar with chart whose inputDateFormat equal "yyyy-MM-dd HH".

```javascript
chart.dateFormatter.inputDateFormat = 'yyyy-MM-dd HH';
```

Then, the bars of the chart in XYChartScrollbar overlapped with each other.

![image](https://user-images.githubusercontent.com/12032476/98432519-5e428100-2102-11eb-89ca-4ee4c55d01ea.png)

How can I avoid the overlapping of the bars ?
Isn't this usage supposed?

**Environment (if applicable)**

 - amCharts version  
amcharts/amcharts4: "^4.10.3"
 - Browser and its version  
Chrome 86.0.4240.183
 - Related frameworks and their versions [e.g. Angular, TypeScript, etc.]  
React

But I reproduced it on [my work in codepen](https://codepen.io/siruku6/pen/bGexaxZ).

**Additional context**

The chart data example is below.

```javascript
[
  {
    'date': '2018-07-01 00:00',
    'open': '136.65',
    'high': '136.96',
    'low': '134.15',
    'close': '136.49'
  }, {
    'date': '2018-07-01 01:00',
    'open': '135.26',
    'high': '135.95',
    'low': '131.50',
    'close': '131.85'
  }, ... , {
    'date': '2018-07-01 03:00',
    'open': '134.94',
    'high': '137.24',
    'low': '132.63',
    'close': '135.03'
  },
]
```

Thank you for your cooperation.

**Comment by jswelsh** [comment]
You may have to just use a generic scrollbar by dropping `scrollbarX.series.push(series);` or create a separate series for the scrollbar. Maybe even one that has formatted dates differently(ie slicing off the times at the end and making it a lineSeries). I personally haven't seen a scrollbar going down to the hour from a series including days within its range, but that's just personal experience from their demos. I've found a few previous closed issues on amCharts GitHub, none were specifically scrolled bar-related though and didn't hint to it being possible. My understanding of scrollbar is that it inherits from the graph, so I don't see why it wouldn't mirror your graph.

[a candle stick with line scrollbar, notice how the scale for the scroll bar is static to month, while the graph scales dynamically](https://www.amcharts.com/demos/candlestick-chart/)

**Comment by siruku6** [comment]
Thank you for your advice!

I understood that your suggestions are
- just use a generic scrollbar  
...It means that the scrollbar does not display any chart.
- create a separate series for the scrollbar  
...It is not candlestick -just simple line chart- .

But I want that the chart displays candlestick chart on scrollbar(preferably...).

---

The example that is one step closer to the expected result is below.

![image](https://user-images.githubusercontent.com/12032476/98436249-40cee080-211d-11eb-97fb-ab16866a96bc.png)

I realized the above result by set `chart.dateFormatter.inputDateFormat = 'yyyy-MM-dd';` and use data without Hour and Minute.

The chart data example is below.

```javascript
[
  {
    'date': '2018-07-01', // without Hour and Minute
    'open': '136.65',
    'high': '136.96',
    'low': '134.15',
    'close': '136.49'
  }, {
    'date': '2018-07-01',
    'open': '135.26',
    'high': '135.95',
    'low': '131.50',
    'close': '131.85'
  }, ... , {
    'date': '2018-07-01',
    'open': '134.94',
    'high': '137.24',
    'low': '132.63',
    'close': '135.03'
  },
]
```

Actually, my data have Hour and Minute, so I hope set hourly candlestick on scrollbarX.

**Comment by zeroin** [comment]
This can be fixed by adding:

`scrollbarX.scrollbarChart.dateFormatter.inputDateFormat = "yyyy-MM-dd HH";`

And we will fix this issue in next release on our side.

**Comment by jswelsh** [comment]
Just to clarify for myself, `chart.dateFormatter.inputDateFormat = 'yyyy-MM-dd HH';` is the intended solution and will be useable later down the road, while `scrollbarX.scrollbarChart.dateFormatter.inputDateFormat = "yyyy-MM-dd HH";` is a work around or is the later the intended solution.

**Comment by siruku6** [comment]
@zeroin @jswelsh 
Thank you very much for your answer, dealing and summary !
I am really glad to hear that.

---

## Adding

 If you would not mind, could you consider to enable using "yyyy-MM-dd HH:mm" and "yyyy-MM-dd HH:mm:ss" ?
Those chart scale is very common in the field of stocks, futures and FX trading.

**Comment by martynasma** [comment]
>  If you would not mind, could you consider to enable using "yyyy-MM-dd HH:mm" and "yyyy-MM-dd HH:mm:ss" ?

After new version release, Scrollbar will automatically use whatever date format you have set on your chart.

**Comment by martynasma** [comment]
Fixed in [4.10.11](https://github.com/amcharts/amcharts4/releases/tag/4.10.11).

## [4.10.11] - 2020-11-13

### Added
- New setting `reverseGeodata` (default: `false`) on `MapChart` and `MapSeries`. amCharts requires polygon coordinates to be in clockwise order. Some map data have them in counter-clockwise. If your custom map appears garbled, try setting this to `true`.
- New global option: `am4core.options.pixelPerfectPrecision` (default: `0`). Precision (number of decimal points) to be used when rounding point x/y for elements that have `pixelPerfect = true` set.

### Fixed
- `XYChartScrollbar` was not inheriting `inputDateFormat` from the chart properly.
- Clicking on a chart on mobile devices could sometimes make them document scroll a bit.
- Accessibility: pressing ENTER while there is a focused element with `showTooltipOn = "hit"` will not display its tooltip.
- RangeSelector (plugin) was not properly removing its elements after `dispose()` it will now auto-dispose when related axis is disposed.
- In some cases a console error could be displayed when using `XYCursor` over chart when data was being updated.
- Very long labels with `wrap = true` may have been wrapping incorrectly in some cases.
- JSON config: `rgb(...)` and `rgba(...)` syntax was not recongnised in `ColorSet`'s `baseColor`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
